### DISCUSSION TITLE
When a backup to WebDav or similar cloud storage happens, are the files first prepared in a temp local dir?

### DISCUSSION THREAD

**Discussion by onecommon-dev** [seedvault-app/seedvault#762] [Q&A] [answered]
Hi,

I have a silly question regarding how Seedvault works internally.
When a backup happens, especially to WebDav or similar cloud location, are the files first prepared in a temp local directory? Or are they simply uploaded directly to the cloud, using in-memory buffers?
My reason for asking is flash storage wear. If the files are uploaded directly, the backup shouldn't incur any wear on the internal flash chip.

I read the code and it looks like it's using in-memory buffers, so it should upload directly without first writing to disk, but I could be wrong.

Thanks in advance!

**Comment by grote** [comment, marked as answer]
We don't duplicate data to disk first, but prepare everything in memory. Storage backends may write to disk first though, I think Nextcloud app still does it for exampe.

**Reply by onecommon-dev** [reply]
Nice, do you know if the WebDav backend writes to disk?

**Reply by grote** [reply]
What is currently called "WebDAV Cloud (beta)" is a built-in backend which doesn't write to disk. It is the recommended way to use all WebDAV based storage locations.

**Reply by onecommon-dev** [reply]
It seems the version of Seedvault that comes with LineageOS 21 has WebDAV Cloud (beta), but the one that's in e/OS/ 2.4 doesn't, and it only has DAVx5

**Reply by grote** [reply]
Yeah, I don't know which version of Seedvault various ROM versions ship. That's up to them, but I believe DavX5 also isn't writing files to disk first.

**Reply by onecommon-dev** [reply]
Thank you very much for the clarification!
